foundations/html_css/html-foundations/commit-messages.md

```diff
@@ -0,0 +1,44 @@
+### Introduction
+
+This lesson will explain the importance of good commit messages, how to write them, when to commit and why having a history of good commits is so important!
+
+### Learning Outcomes
+
+By the end of this lesson, you should be able to:
+
+- Write a meaningful commit message
+- Understand why meaningful commit messages are important
+- Have a good understanding of when to commit
+
+
+### Are commit messages so important they deserve their own lesson?
+
+Yes! Let me give you a quick list of reasons why:
+
+- When you are applying for jobs, employers will look through your projects on GitHub and they will look through your commit history. Having good commits as a novice developer will make you stand apart. 
+- Having a good commit message history will allow you (or other developers working on your code) to quickly see what changes were made and why. This is useful if a bug is found in the code that needs to be fixed!
+- Having a good commit message history will also be helpful if you come back to a project you were working on after stepping away from it for a while. You likely won't remember your thought process and changes made when initially writing out the code. 
+
+### How to Write a Good Commit Message
+
+This article, [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/),  covers all the main bases on how to write good commit messages. The whole article is great and informative but the meat of the article is “The seven rules of a great commit message.”
+
+### When to Commit
+
+A good way to view a commit is like a “snapshot” of your code at the moment that commit was made. That version of your code up to that point will be saved for you to revert back to or look back at. 
+
+When writing code, it's considered best practice to commit early and often. Commit every time you have a meaningful change in the code. This will create a timeline of your progress and show that your finished code didn't appear out of nowhere.
+This means, make a commit if you get a line of code you are working on to function like you want it to, fix a typo, fix a bug, or when you are done for the day. 
```

Comment:
I think we need to be careful not to set people up to be committing *too often*. It's a really hard line to draw for beginners. I'm not positive yet what I would want this line to look like, but I feel like this is a few notches into that "too often" territory.

I also think we shouldn't say for people to just commit when they're done for the day, unless they're actually at a good commit point i.e. the app isn't in a currently non-running state.

If you'd like me to come up with a proposal for how I think we should word this just let me know, also more than happy to try and workshop something.

Reply:
Yeah, 28 is fine, GitHub just automatically selects a handful of lines to show as context when leaving a comment about a specific line.

I like that suggestion more. I think I would just want to also see something added stating that as the student gets more experienced they will get a better feel for what sort of things need to be committed through experience.